### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by Tamunokuro**
- Write unit tests for all classes

### CODE REVIEW

**Review by Davidosky007 — COMMENTED** [review]
Hi @Tamunokuro @SSEKPIUS 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't show a contribution from your partner @SSEKPIUS as this is a paired programming activity your partner is to contribute to this project with code, Kindly work with your partner on this project 🙏  and if you have been given permission to work solo by the student success kindly attach prove to your PR for the next CR to see Thanks  

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/books_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    book.add_rental(date, person)
+    expect(book.rentals.length).to be 0
+  end
+
+  it 'Should have a title and author' do
+    book = Book.new('The Great Gatsby', 'F. Scott Fitzgerald')
+    expect(book.title).to eq('The Great Gatsby')
+    expect(book.author).to eq('F. Scott Fitzgerald')
+  end
+end
```
Comment (CollinsTatang) [inline spec/books_spec.rb:24]: Hi @Tamunokuro and @SSEKPIUS 

You have done a great job so far

- [ ] Please all your rspec test results to failures, kindly fix the test and make it all pass the test before you submit it for the next review.
**Check The Test Result Below**
test

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @Tamunokuro and @SSEKPIUS , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required changes 

- [ ] Pease update the `Author2` section on your README file.

It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job.
_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/books_spec.rb (hunk 1 above)
Comment (SSEKPIUS) [reply spec/books_spec.rb:24]: Screenshot 2023-02-24 at 9 20 05 AM

@CollinsTatang , hope your not missing something

Comment (CollinsTatang) [reply spec/books_spec.rb:24]: Hello @Tamunokuro 
Can we have a call

Comment (danmainah) [reply spec/books_spec.rb:24]: @team your tests are passing, great work

**Review by danmainah — APPROVED** [review]
Hi @Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used

## Optional suggestions

- [ ] @team it would be nice if you added how to run tests in your readme file.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/books_spec.rb (hunk 1 above)
Comment (CollinsTatang) [reply spec/books_spec.rb:24]: Sorry @SSEKPIUS and @Tamunokuro 
Your tests are all passing, I had to make some updates on my end.
I am very sorry for the delay in your project approval

Happy Weekend 

Thanks

Comment (SSEKPIUS) [reply spec/books_spec.rb:24]: Thank you @CollinsTatang  and @danmainah

**Pull request merged**